logger/Include/LogQueue.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <atomic>
+#include <cassert>
+#include <condition_variable>
+#include <queue>
+#include <mutex>
+#include <optional>
+
+template<typename T>
+class LogQueue
+{
+public:
+
+	void Push(T value)
+	{
+		assert(!is_closed);
+		std::lock_guard guard(mutex);
+		buffer.push(std::move(value));
+		is_not_empty.notify_one();
+	}
+
+
+	std::optional<T> Pop()
+	{
+		std::unique_lock lock(mutex);
+		while (buffer.empty())
+		{
+			if (is_closed) return std::nullopt;
```

Comment:
Try to stick to using braces for one liner if statements. It is a QoL because you make sure you won't make any issue when playing with code around it and can save you a headache.

```suggestion
			if (is_closed) {
			    return std::nullopt;
			}
```